include/solvers/PreconditionCG.hpp

```diff
@@ -90,7 +95,7 @@ ITERATIVE_NAMESPACE_OPEN
                 // Solution is: finding the exact Cholesky decomposition, except that any entry is set to zero
                 // if the corresponding entry in A is also zero.
 
-                MatrixType L, K;
+                MatrixType L(A.rows(),A.columns()), K(A.rows(),A.columns());
```

Comment:
This allocates memory for the `K` matrix that is released right away in the assignment below. Is that intentional?